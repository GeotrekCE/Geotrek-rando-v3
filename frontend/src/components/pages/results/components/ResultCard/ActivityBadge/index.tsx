import React from 'react';
import styled from 'styled-components';

import { GenericIconProps } from 'components/Icons/types';
import { colorPalette, getSpacing } from 'stylesheet';

interface Props {
  icon: React.FC<GenericIconProps>;
  className?: string;
}

export const ActivityBadge: React.FC<Props> = ({ icon: Icon, className }) => {
  return (
    <Container className={className}>
      <Icon size={21} />
    </Container>
  );
};

const Container = styled.div`
  width: ${getSpacing(8)};
  height: ${getSpacing(8)};
  border-radius: 50%;

  display: grid;
  place-items: center;

  border: 2px solid ${colorPalette.white};
  background-color: ${colorPalette.primary1};
  color: ${colorPalette.white};
`;
