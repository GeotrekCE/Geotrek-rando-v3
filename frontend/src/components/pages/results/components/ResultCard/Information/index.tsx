import React from 'react';

import { GenericIconProps } from 'components/Icons/types';
import { colorPalette, getSpacing, typography } from 'stylesheet';
import styled from 'styled-components';

interface Props {
  icon: React.FC<GenericIconProps>;
  children: React.ReactNode;
}

export const Information: React.FC<Props> = ({ icon: Icon, children }) => {
  return (
    <Container>
      <Icon size={24} />
      <InformationText>{children}</InformationText>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  align-items: center;
  color: ${colorPalette.primary1};
`;

const InformationText = styled.span`
  margin-left: ${getSpacing(2)};
  ${typography.small}
  color: ${colorPalette.darkPurple}
`;
