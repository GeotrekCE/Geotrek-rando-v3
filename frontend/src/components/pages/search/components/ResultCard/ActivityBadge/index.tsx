import React from 'react';
import styled from 'styled-components';
import SVG from 'react-inlinesvg';

import { colorPalette, getSpacing } from 'stylesheet';

interface Props {
  iconUri: string;
  className?: string;
}

export const ActivityBadge: React.FC<Props> = ({ iconUri, className }) => {
  return (
    <Container className={className}>
      <SVG src={iconUri} className="w-5" />
    </Container>
  );
};

const Container = styled.div`
  width: ${getSpacing(8)};
  height: ${getSpacing(8)};
  border-radius: 50%;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  border: 2px solid ${colorPalette.white};
  background-color: ${colorPalette.primary1};
  color: ${colorPalette.white};
`;
