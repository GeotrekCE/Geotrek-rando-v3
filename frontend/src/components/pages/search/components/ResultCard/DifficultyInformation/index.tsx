import React from 'react';
import SVG from 'react-inlinesvg';

import { colorPalette, getSpacing, typography } from 'stylesheet';
import styled from 'styled-components';

interface Props {
  iconUri: string;
  children: React.ReactNode;
  className?: string;
}

export const DifficultyInformation: React.FC<Props> = ({ iconUri, children, className = '' }) => {
  return (
    <div className={`flex items-center text-primary1 ${className} `}>
      <SVG src={iconUri} className="desktop:w-6" />
      <InformationText>{children}</InformationText>
    </div>
  );
};

const InformationText = styled.span`
  margin-left: ${getSpacing(2)};
  ${typography.small}
  color: ${colorPalette.darkPurple}
`;
