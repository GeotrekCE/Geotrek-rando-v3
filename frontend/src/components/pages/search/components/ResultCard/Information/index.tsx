export { LocalIconInformation } from './LocalIconInformation';
export { RemoteIconInformation } from './RemoteIconInformation';
