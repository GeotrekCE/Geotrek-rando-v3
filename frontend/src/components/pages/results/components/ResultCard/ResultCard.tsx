import React from 'react';
import styled, { css } from 'styled-components';

import { borderRadius, colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';
import { flexGap } from 'services/cssHelpers';

import { Chip } from 'components/Chip';
import { Button } from 'components/Button';

import { Clock } from 'components/Icons/Clock';
import { CodeBrackets } from 'components/Icons/CodeBrackets';
import { TrendingUp } from 'components/Icons/TrendingUp';
import { Walking } from 'components/Icons/Walking';

import { Information } from './Information';
import { ActivityBadge as RawActivityBadge } from './ActivityBadge';

export const ResultCard: React.FC = () => {
  return (
    <Container>
      <ImageContainer>
        <ActivityBadge icon={Walking} />
      </ImageContainer>
      <DetailsContainer>
        <DetailsLayout>
          <Place>Saint-Etienne-du-Valdonnez</Place>
          <Title>Balade au pays des menhirs</Title>
          <TagContainer>
            <TagLayout>
              <Chip>En famille</Chip>
              <Chip>Ciel étoilé</Chip>
              <Chip>En famille</Chip>
            </TagLayout>
          </TagContainer>
          <InformationContainer>
            <InformationLayout>
              <Information icon={Clock}>2h</Information>
              <Information icon={CodeBrackets}>5km</Information>
              <Information icon={TrendingUp}>+360m</Information>
            </InformationLayout>
          </InformationContainer>
        </DetailsLayout>
        <BookingButtonContainer>
          <Button>Réserver</Button>
        </BookingButtonContainer>
      </DetailsContainer>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;

  margin: ${getSpacing(4)};

  border-radius: ${borderRadius.card};
  overflow: hidden;

  ${desktopOnly(
    css`
      flex-direction: row;
      margin: ${getSpacing(6)};
    `,
  )}
`;

const ImageContainer = styled.div`
  height: ${getSpacing(31)};
  width: 100%;

  background-image: url('images/hiking-cover.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;

  position: relative;

  ${desktopOnly(
    css`
      height: auto;
      max-width: ${getSpacing(56)};
      flex-shrink: 0;
    `,
  )}
`;

const DetailsContainer = styled.div`
  display: flex;

  padding: ${getSpacing(4)};

  border: 1px solid ${colorPalette.greySoft};
  border-top: none;
  border-radius: 0 0 ${borderRadius.card} ${borderRadius.card};

  ${desktopOnly(
    css`
      padding: ${getSpacing(6)};

      border: 1px solid ${colorPalette.greySoft};
      border-left: none;
      border-radius: 0 ${borderRadius.card} ${borderRadius.card} 0;
    `,
  )}
`;

const DetailsLayout = styled.div`
  display: flex;
  flex-direction: column;
`;

const BookingButtonContainer = styled.div`
  margin-left: ${getSpacing(4)};
  display: none;

  ${desktopOnly(
    css`
      display: block;
    `,
  )}
`;

const Place = styled.span`
  ${typography.small}
`;

const Title = styled.span`
  margin-top: ${getSpacing(1)};

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;

  ${typography.h1}
  color: ${colorPalette.primary1};

  ${desktopOnly(
    css`
      text-overflow: clip;
      white-space: normal;
    `,
  )}
`;

const TagContainer = styled.div`
  display: flex;
  margin-top: ${getSpacing(2)};

  ${desktopOnly(
    css`
      margin-top: ${getSpacing(4)};
    `,
  )}
`;

const TagLayout = styled.div`
  ${flexGap(getSpacing(2))}

  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;

const InformationContainer = styled.div`
  margin-top: ${getSpacing(4)};
`;

const InformationLayout = styled.div`
  ${flexGap(getSpacing(4))}
`;

const ActivityBadge = styled(RawActivityBadge)`
  position: absolute;
  top: ${getSpacing(4)};
  left: ${getSpacing(4)};
`;
