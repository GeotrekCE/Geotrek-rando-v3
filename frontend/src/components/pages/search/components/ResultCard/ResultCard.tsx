import { Altitude } from 'components/Icons/Altitude';
import { Calendar } from 'components/Icons/Calendar';
import { Height } from 'components/Icons/Height';
import { Modal } from 'components/Modal';
import { DetailsCoverCarousel } from 'components/pages/details/components/DetailsCoverCarousel';
import React, { useContext } from 'react';
import styled, { css } from 'styled-components';
import { FormattedMessage, useIntl } from 'react-intl';

import { borderRadius, colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';
import { flexGap } from 'services/cssHelpers';

import { Chip } from 'components/Chip';
import { Button } from 'components/Button';
import { Link } from 'components/Link';
import { LocalIconInformation, RemoteIconInformation } from 'components/Information';
import { TouristicContentDetailsType } from 'modules/touristicContent/interface';

import { Clock } from 'components/Icons/Clock';
import { CodeBrackets } from 'components/Icons/CodeBrackets';
import { TrendingUp } from 'components/Icons/TrendingUp';
import { ListAndMapContext } from 'modules/map/ListAndMapContext';

import { Attachment } from '../../../../../modules/interface';
import { dataUnits } from '../../../../../modules/results/adapter';
import { ResultCardCarousel } from './ResultCardCarousel';

interface BaseProps {
  id: string;
  hoverId: string | null;
  place: string | null;
  title: string;
  tags: string[];
  thumbnailUris: string[];
  redirectionUrl: string;
  attachments?: Attachment[];
  badgeIconUri?: string;
  badgeName?: string;
  className?: string;
}

interface TrekProps extends BaseProps {
  type: 'TREK';
  informations: {
    duration: string | null;
    distance: string;
    elevation: string;
    difficulty?: { label: string; pictogramUri: string } | null;
    reservationSystem: number | null;
  };
}

interface TouristicContentProps extends BaseProps {
  type: 'TOURISTIC_CONTENT';
  informations: TouristicContentDetailsType[];
}

interface OutdoorSiteProps extends BaseProps {
  type: 'OUTDOOR_SITE';
  informations: TouristicContentDetailsType[];
}

interface OutdoorCourseProps extends BaseProps {
  type: 'OUTDOOR_COURSE';
  informations: {
    duration: string | null;
    elevation?: string | null;
    maxElevation?: number;
    height: string | null;
    length: string | null;
  };
}

interface TouristicEventProps extends BaseProps {
  type: 'TOURISTIC_EVENT';
  informations: Record<any, any>;
}

const isTrek = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is TrekProps => content.type === 'TREK';

const isOutdoorCourse = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is OutdoorCourseProps => content.type === 'OUTDOOR_COURSE';

const isTouristicEvent = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is TouristicEventProps => content.type === 'TOURISTIC_EVENT';

export const ResultCard: React.FC<
  (
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps
  ) & { asColumn?: boolean }
> = props => {
  const {
    id,
    hoverId,
    place,
    title,
    tags,
    thumbnailUris,
    attachments,
    badgeIconUri,
    badgeName,
    className,
    redirectionUrl,
    type,
    asColumn,
  } = props;
  const { setHoveredCardId } = useContext(ListAndMapContext);

  const intl = useIntl();

  return (
    <Container
      onMouseEnter={() => {
        setHoveredCardId(hoverId);
      }}
      onMouseLeave={() => {
        setHoveredCardId(null);
      }}
      className={className}
      id="result_card"
      asColumn={asColumn}
    >
      <Modal>
        {({ isFullscreen, toggleFullscreen }) => (
          <>
            {isFullscreen && attachments && attachments.length > 0 && (
              <DetailsCoverCarousel attachments={attachments} />
            )}
            {(!isFullscreen || !attachments) && (
              <ResultCardCarousel
                asColumn={asColumn}
                type={type}
                thumbnailUris={thumbnailUris}
                iconUri={badgeIconUri}
                iconName={badgeName as string}
                onClickImage={attachments ? toggleFullscreen : undefined}
              />
            )}
          </>
        )}
      </Modal>

      <Link href={redirectionUrl} testId={`Link-ResultCard-${id}`} className="w-full">
        <DetailsContainer>
          <DetailsLayout>
            {place !== null && <Place>{place}</Place>}

            <Title>{title}</Title>

            <TagContainer>
              <TagLayout>
                {tags
                  .filter(tag => tag !== null && Number(tag?.length) > 0)
                  .map(tag => (
                    <Chip key={tag}>{tag}</Chip>
                  ))}
              </TagLayout>
            </TagContainer>
            {isTouristicEvent(props) && (
              <InformationContainer>
                {props.informations.date && (
                  <LocalIconInformation icon={Calendar}>
                    {props.informations.date.beginDate === props.informations.date.endDate ? (
                      <FormattedMessage
                        id={'dates.singleDate'}
                        values={{ date: intl.formatDate(props.informations.date.beginDate) }}
                      />
                    ) : (
                      <FormattedMessage
                        id={'dates.multipleDates'}
                        values={{
                          beginDate: intl.formatDate(props.informations.date.beginDate),
                          endDate: intl.formatDate(props.informations.date.endDate),
                        }}
                      />
                    )}
                  </LocalIconInformation>
                )}
              </InformationContainer>
            )}
            {isOutdoorCourse(props) && (
              <InformationContainer>
                <InformationLayout>
                  {props.informations.duration && (
                    <LocalIconInformation icon={Clock}>
                      {props.informations.duration}
                    </LocalIconInformation>
                  )}
                  {props.informations.elevation && (
                    <LocalIconInformation icon={TrendingUp}>
                      {props.informations.elevation}
                    </LocalIconInformation>
                  )}
                  {props.informations.maxElevation && (
                    <LocalIconInformation icon={Altitude}>
                      {props.informations.maxElevation}
                      {dataUnits.distance}
                    </LocalIconInformation>
                  )}
                  {props.informations.length && (
                    <LocalIconInformation icon={CodeBrackets}>
                      {props.informations.length}
                    </LocalIconInformation>
                  )}
                  {props.informations.height && (
                    <LocalIconInformation icon={Height}>
                      {props.informations.height}
                    </LocalIconInformation>
                  )}
                </InformationLayout>
              </InformationContainer>
            )}
            {isTrek(props) && (
              <InformationContainer>
                <InformationLayout>
                  {props.informations.difficulty && (
                    <RemoteIconInformation iconUri={props.informations.difficulty.pictogramUri}>
                      {props.informations.difficulty.label}
                    </RemoteIconInformation>
                  )}
                  {props.informations.duration && (
                    <LocalIconInformation icon={Clock}>
                      {props.informations.duration}
                    </LocalIconInformation>
                  )}
                  {props.informations.distance && (
                    <LocalIconInformation icon={CodeBrackets}>
                      {props.informations.distance}
                    </LocalIconInformation>
                  )}
                  {props.informations.elevation && (
                    <LocalIconInformation icon={TrendingUp} className="desktop:flex hidden">
                      {props.informations.elevation}
                    </LocalIconInformation>
                  )}
                </InformationLayout>
              </InformationContainer>
            )}
            {!isTrek(props) && !isOutdoorCourse(props) && !isTouristicEvent(props) && (
              <InformationContainer>
                {props.informations.map(
                  ({ label, values }) =>
                    values.length > 0 && (
                      <div key={label} className="text-greyDarkColored">
                        <span className="font-bold">{`${label} : `}</span>
                        {values.join(', ')}
                      </div>
                    ),
                )}
              </InformationContainer>
            )}
          </DetailsLayout>
          {isTrek(props) &&
            props.informations.reservationSystem !== null &&
            false && ( // we disable this button because the booking behaviour is not implemented yet
              <BookingButtonContainer>
                <Button>
                  <FormattedMessage id="search.book" />
                </Button>
              </BookingButtonContainer>
            )}
        </DetailsContainer>
      </Link>
    </Container>
  );
};

const Container = styled.div<{ asColumn?: boolean }>`
  display: flex;
  flex: auto;
  flex-direction: column;
  cursor: pointer;
  border: 1px solid ${colorPalette.greySoft.DEFAULT};
  transition: all 500ms;
  &:hover {
    border-color: ${colorPalette.blackSemiTransparent};
  }
  border-radius: ${borderRadius.card};
  overflow: hidden;
  // Fix for overflow hidden with border radius in Safari, see https://gist.github.com/ayamflow/b602ab436ac9f05660d9c15190f4fd7b
  -webkit-mask-image: -webkit-radial-gradient(white, black);
  mask-image: radial-gradient(white, black);

  align-items: stretch;

  ${({ asColumn }) =>
    asColumn !== true &&
    desktopOnly(
      css`
        flex-direction: row;
      `,
    )}
`;

const DetailsContainer = styled.div`
  display: flex;
  width: 100%;

  padding: ${getSpacing(4)};

  ${desktopOnly(
    css`
      padding: ${getSpacing(6)};
    `,
  )}
`;

const DetailsLayout = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const BookingButtonContainer = styled.div`
  margin-left: ${getSpacing(4)};
  display: none;

  ${desktopOnly(
    css`
      display: block;
    `,
  )}
`;

const Place = styled.span`
  color: ${colorPalette.greyDarkColored};
  ${typography.small}
`;

const Title = styled.span`
  margin-top: ${getSpacing(1)};

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;

  ${typography.h3}
  color: ${colorPalette.primary1};

  ${desktopOnly(
    css`
      text-overflow: clip;
      white-space: normal;
    `,
  )}
`;

const TagContainer = styled.div`
  display: flex;
  margin-top: ${getSpacing(2)};

  ${desktopOnly(
    css`
      margin-top: ${getSpacing(4)};
    `,
  )}
`;

const TagLayout = styled.div`
  ${flexGap(getSpacing(2))}

  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;

const InformationContainer = styled.div`
  margin-top: ${getSpacing(4)};
`;

const InformationLayout = styled.div`
  ${flexGap(getSpacing(2))}
  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;
