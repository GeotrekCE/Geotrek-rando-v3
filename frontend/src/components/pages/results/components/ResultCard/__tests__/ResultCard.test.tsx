import { render } from 'services/testing/reactTestingLibraryWrapper';

import { Walking } from 'components/Icons/Walking';

import { ResultCard } from '../ResultCard';

test('AAU, I can see a ResultCard', () => {
  const component = render(
    <ResultCard
      activityIcon={Walking}
      place="Saint-Etienne-du-Valdonnez"
      title="Balade au pays des menhirs"
      tags={['En famille', 'Ciel étoilé', 'Beau paysage']}
      informations={{ duration: '2h', distance: '5km', elevation: '+360m', difficulty: '' }}
    />,
  );

  expect(component).toMatchSnapshot();
});
