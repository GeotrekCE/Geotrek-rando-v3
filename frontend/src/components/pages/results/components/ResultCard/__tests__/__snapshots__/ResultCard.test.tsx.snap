// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AAU, I can see a ResultCard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ResultCard__Container-cct1km-0 bhOJjL"
      >
        <div
          class="ResultCard__ImageContainer-cct1km-1 hBcqhI"
        >
          <div
            class="ActivityBadge__Container-sc-11f422r-0 hQCoJE ResultCard__ActivityBadge-cct1km-11 eYtRJc"
          >
            <svg
              fill="none"
              height="21"
              viewBox="0 0 24 25"
              width="21"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.102 10.819l-2.777-1.554-2.292-3.643c-.374-.476-.96-.815-1.606-.815-.51 0-.934.229-1.273.535L6.689 8.908c-.136.102-.196.263-.23.433l-.433 3.898c-.033.068-.025.094-.025.128.034.475.416.874.892.84.441 0 .772-.373.84-.815l.357-3.337 1.197-1.223-1.07 9.375-2.012 4.433c-.068.17-.136.382-.102.586 0 .713.62 1.274 1.3 1.274.509 0 .908-.323 1.146-.764l2.14-4.79c.034-.102.203-.425.203-.56l.255-2.319 2.217 7.516c.17.543.67.917 1.248.917.713-.034 1.282-.637 1.248-1.35 0-.102-.025-.153-.025-.255L12.752 12.45l.408-3.389.866 1.376a.728.728 0 00.23.23l3.031 1.68a.904.904 0 00.408.103.894.894 0 00.84-.918.954.954 0 00-.433-.713zm-6.445-6.42c1.053-.034 1.893-.926 1.86-2.013-.034-1.053-.918-1.919-1.937-1.885a1.937 1.937 0 00-1.86 1.987c.034 1.087.918 1.91 1.937 1.91z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="ResultCard__DetailsContainer-cct1km-2 hAMaUS"
        >
          <div
            class="ResultCard__DetailsLayout-cct1km-3 bPbkur"
          >
            <span
              class="ResultCard__Place-cct1km-5 cDmQiy"
            >
              Saint-Etienne-du-Valdonnez
            </span>
            <span
              class="ResultCard__Title-cct1km-6 VgvlK"
            >
              Balade au pays des menhirs
            </span>
            <div
              class="ResultCard__TagContainer-cct1km-7 dBMGrR"
            >
              <div
                class="ResultCard__TagLayout-cct1km-8 byiiky"
              >
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
                >
                  <span>
                    En famille
                  </span>
                </div>
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
                >
                  <span>
                    Ciel étoilé
                  </span>
                </div>
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
                >
                  <span>
                    Beau paysage
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ResultCard__InformationContainer-cct1km-9 gVihWQ"
            >
              <div
                class="ResultCard__InformationLayout-cct1km-10 inATfC"
              >
                <div
                  class="Information__Container-sc-1mw7hhq-0 kcyBZH"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M12 6v6l4 2"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                  >
                    2h
                  </span>
                </div>
                <div
                  class="Information__Container-sc-1mw7hhq-0 kcyBZH"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 18l6-6-6-6M8 6l-6 6 6 6"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                  >
                    5km
                  </span>
                </div>
                <div
                  class="Information__Container-sc-1mw7hhq-0 kcyBZH"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M23 6l-9.5 9.5-5-5L1 18"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M17 6h6v6"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                  >
                    +360m
                  </span>
                </div>
                <div
                  class="Information__Container-sc-1mw7hhq-0 kcyBZH ResultCard__DifficultyInformation-cct1km-12 kUjqTK"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M23.52 19.68a3.851 3.851 0 01-3.84 3.84H4.32a3.851 3.851 0 01-3.84-3.84V4.32A3.851 3.851 0 014.32.48h15.36a3.851 3.851 0 013.84 3.84v15.36z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ResultCard__BookingButtonContainer-cct1km-4 eRpIfc"
          >
            <button
              class="Button-sc-1n2k9lu-0 irBQYP"
            >
              results.book
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ResultCard__Container-cct1km-0 bhOJjL"
    >
      <div
        class="ResultCard__ImageContainer-cct1km-1 hBcqhI"
      >
        <div
          class="ActivityBadge__Container-sc-11f422r-0 hQCoJE ResultCard__ActivityBadge-cct1km-11 eYtRJc"
        >
          <svg
            fill="none"
            height="21"
            viewBox="0 0 24 25"
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.102 10.819l-2.777-1.554-2.292-3.643c-.374-.476-.96-.815-1.606-.815-.51 0-.934.229-1.273.535L6.689 8.908c-.136.102-.196.263-.23.433l-.433 3.898c-.033.068-.025.094-.025.128.034.475.416.874.892.84.441 0 .772-.373.84-.815l.357-3.337 1.197-1.223-1.07 9.375-2.012 4.433c-.068.17-.136.382-.102.586 0 .713.62 1.274 1.3 1.274.509 0 .908-.323 1.146-.764l2.14-4.79c.034-.102.203-.425.203-.56l.255-2.319 2.217 7.516c.17.543.67.917 1.248.917.713-.034 1.282-.637 1.248-1.35 0-.102-.025-.153-.025-.255L12.752 12.45l.408-3.389.866 1.376a.728.728 0 00.23.23l3.031 1.68a.904.904 0 00.408.103.894.894 0 00.84-.918.954.954 0 00-.433-.713zm-6.445-6.42c1.053-.034 1.893-.926 1.86-2.013-.034-1.053-.918-1.919-1.937-1.885a1.937 1.937 0 00-1.86 1.987c.034 1.087.918 1.91 1.937 1.91z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="ResultCard__DetailsContainer-cct1km-2 hAMaUS"
      >
        <div
          class="ResultCard__DetailsLayout-cct1km-3 bPbkur"
        >
          <span
            class="ResultCard__Place-cct1km-5 cDmQiy"
          >
            Saint-Etienne-du-Valdonnez
          </span>
          <span
            class="ResultCard__Title-cct1km-6 VgvlK"
          >
            Balade au pays des menhirs
          </span>
          <div
            class="ResultCard__TagContainer-cct1km-7 dBMGrR"
          >
            <div
              class="ResultCard__TagLayout-cct1km-8 byiiky"
            >
              <div
                class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
              >
                <span>
                  En famille
                </span>
              </div>
              <div
                class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
              >
                <span>
                  Ciel étoilé
                </span>
              </div>
              <div
                class="Chip__ChipContainer-sc-1wajafu-0 gHWJWu"
              >
                <span>
                  Beau paysage
                </span>
              </div>
            </div>
          </div>
          <div
            class="ResultCard__InformationContainer-cct1km-9 gVihWQ"
          >
            <div
              class="ResultCard__InformationLayout-cct1km-10 inATfC"
            >
              <div
                class="Information__Container-sc-1mw7hhq-0 kcyBZH"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M12 6v6l4 2"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                >
                  2h
                </span>
              </div>
              <div
                class="Information__Container-sc-1mw7hhq-0 kcyBZH"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 25"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 18l6-6-6-6M8 6l-6 6 6 6"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                >
                  5km
                </span>
              </div>
              <div
                class="Information__Container-sc-1mw7hhq-0 kcyBZH"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 25"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M23 6l-9.5 9.5-5-5L1 18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M17 6h6v6"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                >
                  +360m
                </span>
              </div>
              <div
                class="Information__Container-sc-1mw7hhq-0 kcyBZH ResultCard__DifficultyInformation-cct1km-12 kUjqTK"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 25"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M23.52 19.68a3.851 3.851 0 01-3.84 3.84H4.32a3.851 3.851 0 01-3.84-3.84V4.32A3.851 3.851 0 014.32.48h15.36a3.851 3.851 0 013.84 3.84v15.36z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="Information__InformationText-sc-1mw7hhq-1 kpVHAN"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ResultCard__BookingButtonContainer-cct1km-4 eRpIfc"
        >
          <button
            class="Button-sc-1n2k9lu-0 irBQYP"
          >
            results.book
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
